village/src/App.js

```diff
@@ -1,24 +1,69 @@
-import React, { Component } from 'react';
+import React, { Component } from "react";
 
-import './App.css';
-import SmurfForm from './components/SmurfForm';
-import Smurfs from './components/Smurfs';
+import "./App.css";
+import SmurfForm from "./components/SmurfForm";
+import Smurfs from "./components/Smurfs";
+import NavBar from "./components/NavBar";
+import axios from "axios";
+import { Route } from "react-router-dom";
 
 class App extends Component {
   constructor(props) {
     super(props);
     this.state = {
       smurfs: [],
+      updated: false
     };
   }
+
   // add any needed code to ensure that the smurfs collection exists on state and it has data coming from the server
   // Notice what your map function is looping over and returning inside of Smurfs.
   // You'll need to make sure you have the right properties on state and pass them down to props.
+
+  componentDidMount() {
+    axios
+      .get(`http://localhost:3333/smurfs`)
+      .then(res => {
+        this.setState({
+          smurfs: res.data
+        });
+      })
+      .catch(err => {
+        console.log(err);
+      });
+  }
+
+  deleteSmurf = (e, id) => {
+    e.preventDefault();
+    axios
+      .delete(`http://localhost:3333/smurfs/${id}`)
+      .then(res => {
+        this.setState({ smurfs: res.data });
+        this.props.history.push("/");
```

Comment:
Line 42 is unnecessary, we're already on `/` and we're setting state so the component will re-render